app/models/mixins/external_url_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ExternalUrlMixin
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :external_urls, :as => :resource, :dependent => :destroy
+  end
+
+  def external_url=(url, user = User.current_user)
```

Comment:
Sorry, one other thing here @martinpovolny. How is this second parameter supposed to be used? I would have thought we might send a particular user in specs rather than relying on `User.current_user`, but we are not doing that.

So are we really expecting some caller to use this parameter with `.send` or something?

Reply:
https://github.com/ManageIQ/manageiq-automation_engine/pull/328/files#diff-a7b878b75a3f937464b0c265c59aba39R4

Reply:
yes, I made the model function with the same name as the automate method so that it's easy to relate.

Given on the model (unlike in automate) the user is being added, one needs to call send.

Is that a problem?

Reply:
Not a problem, just making sure we had a good reason. Typically I would say to not use the assignment style if we're using more than one parameter, but if you're doing it for consistency then I'm okay with that.